diff --git a/Source/cmakemain.cxx b/Source/cmakemain.cxx
index d662a9abca..c722ab7582 100644
--- a/Source/cmakemain.cxx
+++ b/Source/cmakemain.cxx
@@ -686,6 +686,15 @@ int do_open(int ac, char const* const* av)
 
 int main(int ac, char const* const* av)
 {
+  // Reset LD_LIBRARY_PATH if we are running in an AppDir
+  char* appimage_var = getenv("MESON_AppRun_LD_LIBRARY_PATH");
+  if (appimage_var) {
+    size_t len = 18 + strlen(appimage_var);
+    char* newenv = reinterpret_cast<char*>(malloc(len * sizeof(char)));
+    snprintf(newenv, len, "LD_LIBRARY_PATH=%s", appimage_var);
+    putenv(newenv);
+  }
+
   cmSystemTools::EnsureStdPipes();
 #if defined(_WIN32) && !defined(CMAKE_BOOTSTRAP)
   // Replace streambuf so we can output Unicode to console
